File: leetcode/0088-merge-sorted-array/solution.py
class Solution:
    def merge(self, nums1: list[int], m: int, nums2: list[int], n: int) -> None:
        if (n == 0) :
            nums1.sort()
        elif (m == 0) :
            for i in range(n) :
                nums1[i] = nums2[i]
            nums1.sort()
        else :
            j = 0
            for i in range(m, m+n) :
                nums1[i] = nums2[j]
                print(f"i {i}, j {j}")
                j += 1
            nums1.sort()

File: leetcode/0088-merge-sorted-array/Solution.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if (n == 0) {
            Arrays.sort(nums1);
        }
        else if (m == 0) {
            for (int i = 0; i < nums2.length; i++) nums1[i] = nums2[i];
            Arrays.sort(nums1);
        }

        else {
            // m != 0 && n != 0
            int j = 0;
            for (int i = m; i < m+n; i++) { 
                nums1[i] = nums2[j];  
                j++;
            }
            Arrays.sort(nums1);
        }

    }
}